test/ssc_test/save_as_JSON_test.cpp

```diff
@@ -0,0 +1,217 @@
+int np1 = sprintf(inputs_as_JSON1, "%s/test/input_json/pvwatts_merchantplant.json", SSCDIR);
+char inputs_as_JSON2[256] = {};
+int np2 = sprintf(inputs_as_JSON2, "%s/test/input_json/PV_Batt_MP.json", SSCDIR);
+char inputs_as_JSON3[256] = {};
+int np3 = sprintf(inputs_as_JSON3, "%s/test/input_json/PT_MP.json", SSCDIR);
+char inputs_as_JSON4[256] = {};
+int np4 = sprintf(inputs_as_JSON4, "%s/test/input_json/Trough_MP.json", SSCDIR);
+char inputs_as_JSON5[256] = {};
+int np5 = sprintf(inputs_as_JSON5, "%s/test/input_json/Wind_MP.json", SSCDIR);
+
+
+
+TEST(save_as_JSON_test, pvwatts_mechant_plant_read_file_to_string) {
+    std::ifstream test(inputs_as_JSON);
+    std::string str((std::istreambuf_iterator<char>(test)), std::istreambuf_iterator<char>());
+    EXPECT_EQ(str.length(), 5271794);
+}
+
+TEST(save_as_JSON_test_parse, pvwatts_mechant_plant_rapidjson_parse_file) {
+    std::ifstream test(inputs_as_JSON, std::ifstream::binary);
+    rapidjson::Document document;
+    rapidjson::IStreamWrapper isw(test);
+    document.ParseStream(isw);
+    if (document.HasParseError())
+    {
+        // report to the user the failure and their locations in the document.
+        std::cout << document.GetParseError()
+            << "\n";
+    }
+    EXPECT_TRUE(!document.HasParseError());
+}
+TEST(save_as_JSON_test_parse, pvwatts_mechant_plant_rapidjson_parse_file_freadstream) {
+ //   std::ifstream test(inputs_as_JSON, std::ifstream::binary);
+    FILE* fp = fopen(inputs_as_JSON, "rb"); // non-Windows use "r"
+    char readBuffer[65536];
+    rapidjson::FileReadStream is(fp, readBuffer, sizeof(readBuffer));
+    rapidjson::Document document;
+ //   rapidjson::IStreamWrapper isw(test);
+    document.ParseStream(is);
+    if (document.HasParseError())
+    {
+        // report to the user the failure and their locations in the document.
+        std::cout << document.GetParseError()
+            << "\n";
+    }
+    EXPECT_TRUE(!document.HasParseError());
+}
+
+
+TEST(save_as_JSON_test_parse, pv_batt_mechant_plant_rapidjson_parse_file) {
+    std::ifstream test(inputs_as_JSON2, std::ifstream::binary);
+    rapidjson::Document document;
+    rapidjson::IStreamWrapper isw(test);
+    document.ParseStream(isw);
+    if (document.HasParseError())
+    {
+        // report to the user the failure and their locations in the document.
+        std::cout << document.GetParseError()
+            << "\n";
+    }
+    EXPECT_TRUE(!document.HasParseError());
+}
+
+TEST(save_as_JSON_test_parse, pt_mechant_plant_rapidjson_parse_file) {
+    std::ifstream test(inputs_as_JSON3, std::ifstream::binary);
+    rapidjson::Document document;
+    rapidjson::IStreamWrapper isw(test);
+    document.ParseStream(isw);
+    if (document.HasParseError())
+    {
+        // report to the user the failure and their locations in the document.
+        std::cout << document.GetParseError()
+            << "\n";
+    }
+    EXPECT_TRUE(!document.HasParseError());
```

Comment:
Any reason you aren't using EXPECT_FALSE here? That seems more clear to me than EXPECT_TRUE(!...